cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -2145,6 +2419,23 @@ auto merge(raft::resources const& res,
            const cuvs::neighbors::cagra::merge_params& params,
            std::vector<cuvs::neighbors::cagra::index<uint8_t, uint32_t>*>& indices)
   -> cuvs::neighbors::cagra::index<uint8_t, uint32_t>;
+
+auto make_composite_index(const cagra::merge_params& params,
```

Comment:
Should a user ever need to call this? The `merge()` is really the `make_composite_index` isn't it? And the guts of the merge is intentionally kept as implementaiton details. To be clear- I think the factory functions for constructing complex objects are important, but I thought `merge()` was the factory function to accomplish that.